vcs/gitcmd/repo.go

```diff
@@ -161,105 +161,123 @@ func (r *Repository) ResolveTag(name string) (vcs.CommitID, error) {
 	return commitID, nil
 }
 
-// branchCounts returns the behind/ahead commit counts information for branch, against base branch.
-func (r *Repository) branchCounts(branch, base string) (behind, ahead uint, err error) {
-	if err := checkSpecArgSafety(branch); err != nil {
-		return 0, 0, err
-	}
-	if err := checkSpecArgSafety(base); err != nil {
-		return 0, 0, err
-	}
+// stringFilter helps manage a list of string filters.
+type stringFilter struct{ only map[string]struct{} }
```

Comment:
I like it! I will definitely use your version.

Fun fact: this was actually called `branchFilter` initially and have recently renamed it to `stringFilter`, though it'd be easier to pick-up :)